File: Insta Locker App/app/src/main/java/com/example/instalockerappsss/activities/MainActivity.java
package com.example.instalockerappsss.activities;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.app.Dialog;
import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.view.MenuItem;

import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.TextView;

import com.example.instalockerappsss.R;
import com.example.instalockerappsss.fragment.HomeFragment;
import com.example.instalockerappsss.fragment.SettingFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import java.util.Objects;

public class MainActivity extends AppCompatActivity {
    FragmentManager fragmentManager;
    FragmentTransaction fragmentTransaction;

    @Override
    public void onBackPressed() {
        privacy_Dialog();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_main);
        HomeFragment homeFragment = new HomeFragment();
        fragmentManager = getSupportFragmentManager();
        fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.fragment_container, homeFragment, null);
        fragmentTransaction.commit();
        BottomNavigationView bottomNavigationView = (BottomNavigationView) findViewById(R.id.bottom_navigation);
        bottomNavigationView.setItemIconTintList(null);
        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                switch (item.getItemId()) {
                    case R.id.add_user:
                        HomeFragment homeFragment = new HomeFragment();
                        fragmentManager = getSupportFragmentManager();
                        fragmentTransaction = fragmentManager.beginTransaction();
                        fragmentTransaction.replace(R.id.fragment_container, homeFragment, null);
                        fragmentTransaction.commit();
                        break;
                    case R.id.settings:
                        SettingFragment settingFragment = new SettingFragment();
                        fragmentManager = getSupportFragmentManager();
                        fragmentTransaction = fragmentManager.beginTransaction();
                        fragmentTransaction.replace(R.id.fragment_container, settingFragment, null);
                        fragmentTransaction.commit();
                        break;
                }
                return true;
            }
        });
    }
    private void privacy_Dialog() {
        final Dialog dialog = new Dialog(MainActivity.this);
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        dialog.setCancelable(false);
        dialog.setContentView(R.layout.exit_confirm_dialogbox);
        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(android.graphics.Color.TRANSPARENT));
        dialog.getWindow().setLayout(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.WRAP_CONTENT);
        final Button ok = dialog.findViewById(R.id.btn_yes);
        final Button no = dialog.findViewById(R.id.btn_no);
        final Button rate_us = dialog.findViewById(R.id.btn_rateUs);
        rate_us.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                rateApp();
            }
        });
        no.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dialog.dismiss();
            }
        });
        ok.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finishAffinity();
            }
        });
        dialog.show();
    }

    private Intent rateIntentForUrl(String url) {
        Intent intent = null;
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
            intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url + Objects.requireNonNull(MainActivity.this).getPackageName()));
        }
        int flags = Intent.FLAG_ACTIVITY_NO_HISTORY | Intent.FLAG_ACTIVITY_MULTIPLE_TASK;
        if (Build.VERSION.SDK_INT >= 21) {
            flags |= Intent.FLAG_ACTIVITY_NEW_DOCUMENT;
        } else {
            flags |= Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET;
        }
        intent.addFlags(flags);
        return intent;
    }

    private void rateApp() {
        try {
            Intent rateIntent = rateIntentForUrl("market://details?id=");
            startActivity(rateIntent);
        } catch (ActivityNotFoundException e) {
            Intent rateIntent = rateIntentForUrl("https://play.google.com/store/apps/details?id=");
            startActivity(rateIntent);
        }
    }
}

File: Lesson5Lab1/app/src/main/java/com/example/lesson5lab1/MainActivity.kt
package com.example.lesson5lab1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.CheckBox
import android.widget.RadioButton
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val tv_dev_status=findViewById<TextView>(R.id.tv_dev_status);

        findViewById<CheckBox>(R.id.cb_usa).setOnCheckedChangeListener { buttonView, isChecked ->
            if(isChecked){
                Toast.makeText(this,buttonView.text.toString()+" is Correct Answer",Toast.LENGTH_SHORT).show()
            }
        }

        findViewById<CheckBox>(R.id.cb_canada).setOnCheckedChangeListener { buttonView, isChecked ->
            if(isChecked){
                Toast.makeText(this,buttonView.text.toString()+" is Correct Answer",Toast.LENGTH_SHORT).show()
            }
        }

        findViewById<CheckBox>(R.id.cb_china).setOnCheckedChangeListener { buttonView, isChecked ->
            if(isChecked){
                Toast.makeText(this,buttonView.text.toString()+" is not a Correct Answer",Toast.LENGTH_SHORT).show()
            }
        }

        findViewById<RadioButton>(R.id.rb_no).setOnCheckedChangeListener { _, isChecked ->
            if(isChecked){
             tv_dev_status.text="No. not yet!!"
            }
        }
        findViewById<RadioButton>(R.id.rb_yes).setOnCheckedChangeListener { _, isChecked ->
            if(isChecked){
                tv_dev_status.text="Yes. I am!!"
            }
        }

    }
}